// next-client-cookies/dist/server.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var server_exports = {};
__export(server_exports, {
  CookiesProvider: () => CookiesProvider,
  getCookies: () => getCookies
});
module.exports = __toCommonJS(server_exports);
var import_headers = require("next/headers");
var import_react = __toESM(require("react"));
var import_provider = require("./provider");
var import_secure = require("./secure");
const CookiesProvider = /* @__PURE__ */ __name(({ children }) => {
  const secretValue = (0, import_headers.cookies)().getAll();
  return /* @__PURE__ */ import_react.default.createElement(import_provider.SecureCookiesProvider, {
    value: (0, import_secure.storeSecureCookies)(secretValue)
  }, children);
}, "CookiesProvider");
const getCookies = /* @__PURE__ */ __name(() => {
  const org = (0, import_headers.cookies)();
  return {
    get: (name) => name == null ? Object.fromEntries(org.getAll().map((c) => [
      c.name,
      c.value
    ])) : org.get(name)?.value,
    set: (name, value, options) => {
      const pre = org.get(name)?.value;
      org.set(name, value, options && convertCookieAttributes(options));
      return pre;
    },
    remove: (name) => org.delete(name)
  };
}, "getCookies");
const convertCookieAttributes = /* @__PURE__ */ __name((options) => ({
  expires: typeof options.expires === "number" ? options.expires * 864e5 : options.expires,
  path: options.path,
  domain: options.domain,
  secure: options.secure,
  sameSite: options.sameSite?.toLowerCase()
}), "convertCookieAttributes");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CookiesProvider,
  getCookies
});
